core/tests/performance_framework/perf_services.py

```diff
@@ -148,11 +203,14 @@ def get_page_timings_from_cached_session(self, page_url):
         """
```

Comment:
Is there a reason you're doing page_timings and har stuff separately? It seems to me like you are redoing a whole lot of work each time in setting up the test conditions and navigating to the page, which I can imagine causing the tests to be really slow. Why not do both in the same method?